### ISSUE TITLE
Assertion during repair

### ISSUE THREAD

**Issue by glommer** [scylladb/scylla#4357]
This is coming from a user on a PoC of Scylla on slack, opening on their behalf. If more info is needed, I will pursue. Let me know.

[xx.txt](https://github.com/scylladb/scylla/files/2997192/xx.txt)

This repair in the logs  was triggered by using an external tool  https://github.com/BrianGallew/cassandra_range_repair

However the user claims that the very reason they moved to that tool was that `nodetool repair` was crashing the node.

They are okay moving to the Scylla manager, but would like to understand first what is going on (it is also not clear that the manager will not crash)

@asias  - any idea?

decoded backtrace:

```
[Backtrace #0]
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/src/debug/scylla-2.3.1/seastar/util/backtrace.hh:56
seastar::backtrace_buffer::append_backtrace() at /usr/src/debug/scylla-2.3.1/seastar/core/reactor.cc:390
 (inlined by) print_with_backtrace at /usr/src/debug/scylla-2.3.1/seastar/core/reactor.cc:411
seastar::print_with_backtrace(char const*) at /usr/src/debug/scylla-2.3.1/seastar/core/reactor.cc:418
sigabrt_action at /usr/src/debug/scylla-2.3.1/seastar/core/reactor.cc:3938
 (inlined by) operator() at /usr/src/debug/scylla-2.3.1/seastar/core/reactor.cc:3920
 (inlined by) _FUN at /usr/src/debug/scylla-2.3.1/seastar/core/reactor.cc:3916
_L_unlock_13 at funlockfile.c:?
__GI_raise at :?
__GI_abort at :?
__assert_fail_base at :?
__GI___assert_fail at :?
std::pair<nonwrapping_range<dht::token>, nonwrapping_range<dht::token> > nonwrapping_range<dht::token>::split<dht::token_comparator>(dht::token const&, dht::token_comparator&&) const at /usr/src/debug/scylla-2.3.1/./range.hh:543
 (inlined by) range_splitter::next() at /usr/src/debug/scylla-2.3.1/repair/range_split.hh:69
_ZZZZL15repair_cf_rangeR11repair_infoN7seastar13basic_sstringIcjLj15ELb1EEE17nonwrapping_rangeIN3dht5tokenEERKSt6vectorIN3gms12inet_addressESaISA_EEENKUlmE_clEmENKUlRT_RT0_RKT1_RT2_E_clINS1_4gateEbS3_14range_splitterEEDaSH_SJ_SM_SO_ENKUlvE0_clEv at /usr/src/debug/scylla-2.3.1/repair/repair.cc:786
apply<repair_cf_range(repair_info&, seastar::sstring, dht::token_range, const std::vector<gms::inet_address>&)::<lambda(uint64_t)>::<lambda(auto:153&, auto:154&, const auto:155&, auto:156&)> [with auto:153 = seastar::gate; auto:154 = bool; auto:155 = seastar::basic_sstring<char, unsigned int, 15>; auto:156 = range_splitter]::<lambda()>&> at /usr/src/debug/scylla-2.3.1/seastar/core/future.hh:1347
 (inlined by) apply<repair_cf_range(repair_info&, seastar::sstring, dht::token_range, const std::vector<gms::inet_address>&)::<lambda(uint64_t)>::<lambda(auto:153&, auto:154&, const auto:155&, auto:156&)> [with auto:153 = seastar::gate; auto:154 = bool; auto:155 = seastar::basic_sstring<char, unsigned int, 15>; auto:156 = range_splitter]::<lambda()>&> at /usr/src/debug/scylla-2.3.1/seastar/core/future.hh:1373
 (inlined by) do_until<repair_cf_range(repair_info&, seastar::sstring, dht::token_range, const std::vector<gms::inet_address>&)::<lambda(uint64_t)>::<lambda(auto:153&, auto:154&, const auto:155&, auto:156&)> [with auto:153 = seastar::gate; auto:154 = bool; auto:155 = seastar::basic_sstring<char, unsigned int, 15>; auto:156 = range_splitter]::<lambda()>, repair_cf_range(repair_info&, seastar::sstring, dht::token_range, const std::vector<gms::inet_address>&)::<lambda(uint64_t)>::<lambda(auto:153&, auto:154&, const auto:155&, auto:156&)> [with auto:153 = seastar::gate; auto:154 = bool; auto:155 = seastar::basic_sstring<char, unsigned int, 15>; auto:156 = range_splitter]::<lambda()> > at /usr/src/debug/scylla-2.3.1/seastar/core/future-util.hh:503
 (inlined by) operator()<seastar::gate, bool, seastar::basic_sstring<char, unsigned int, 15>, range_splitter> at /usr/src/debug/scylla-2.3.1/repair/repair.cc:968
 (inlined by) apply at /usr/src/debug/scylla-2.3.1/seastar/core/apply.hh:36
 (inlined by) apply<repair_cf_range(repair_info&, seastar::sstring, dht::token_range, const std::vector<gms::inet_address>&)::<lambda(uint64_t)>::<lambda(auto:153&, auto:154&, const auto:155&, auto:156&)>&, seastar::gate, bool, seastar::basic_sstring<char, unsigned int, 15, true>, range_splitter> at /usr/src/debug/scylla-2.3.1/seastar/core/apply.hh:51
 (inlined by) do_with<seastar::gate, bool, const seastar::basic_sstring<char, unsigned int, 15>, range_splitter, repair_cf_range(repair_info&, seastar::sstring, dht::token_range, const std::vector<gms::inet_address>&)::<lambda(uint64_t)>::<lambda(auto:153&, auto:154&, const auto:155&, auto:156&)> > at /usr/src/debug/scylla-2.3.1/seastar/core/do_with.hh:152
operator() at /usr/src/debug/scylla-2.3.1/repair/repair.cc:968
 (inlined by) apply at /usr/src/debug/scylla-2.3.1/seastar/core/apply.hh:36
 (inlined by) apply<repair_cf_range(repair_info&, seastar::sstring, dht::token_range, const std::vector<gms::inet_address>&)::<lambda(uint64_t)>, long unsigned int> at /usr/src/debug/scylla-2.3.1/seastar/core/apply.hh:44
 (inlined by) apply<repair_cf_range(repair_info&, seastar::sstring, dht::token_range, const std::vector<gms::inet_address>&)::<lambda(uint64_t)>, long unsigned int> at /usr/src/debug/scylla-2.3.1/seastar/core/future.hh:1380
 (inlined by) operator()<seastar::future_state<long unsigned int> > at /usr/src/debug/scylla-2.3.1/seastar/core/future.hh:954
 (inlined by) run_and_dispose at /usr/src/debug/scylla-2.3.1/seastar/core/future.hh:413
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/src/debug/scylla-2.3.1/seastar/core/reactor.cc:2620
seastar::reactor::run_some_tasks() at /usr/src/debug/scylla-2.3.1/seastar/core/reactor.cc:3032
seastar::reactor::run_some_tasks() at /opt/scylladb/include/c++/7/chrono:377
 (inlined by) seastar::reactor::run() at /usr/src/debug/scylla-2.3.1/seastar/core/reactor.cc:3179
seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at /usr/src/debug/scylla-2.3.1/seastar/core/app-template.cc:185
main at /usr/src/debug/scylla-2.3.1/main.cc:783
__libc_start_main at ??:?
_start at ??:?
```

**Comment by glommer** [comment]
Scylla version: 2.3.1-0.20181021.336c77166

**Comment by slivne** [comment]
@asias any update on this one ?

**Comment by slivne** [comment]
repair has changed in 3.1 to row_level_repair closing this issue as we will not fix issues in open source 2.3 related to partition level repair
